src/test/java/api/requests/scenarios/LoanDueDatesAfterRecallTests.java

```diff
@@ -0,0 +1,398 @@
+import java.time.ZonedDateTime;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.folio.circulation.domain.policy.DueDateManagement;
+import org.folio.circulation.domain.policy.Period;
+import org.folio.circulation.support.ClockManager;
+import org.folio.circulation.support.http.client.IndividualResource;
+import org.folio.circulation.support.http.client.Response;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeConstants;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.ISODateTimeFormat;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import api.support.APITests;
+import api.support.builders.LoanBuilder;
+import api.support.builders.LoanPolicyBuilder;
+import api.support.builders.ServicePointBuilder;
+import io.vertx.core.json.JsonObject;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import junitparams.converters.Nullable;
+import junitparams.naming.TestCaseName;
+
+/**
+ * Notes:<br>
+ *  MGD = Minimum guaranteed due date<br>
+ *  RD = Recall due date<br>
+ * 
+ * @author mreno
+ *
+ */
+@RunWith(JUnitParamsRunner.class)
+public class LoanDueDatesAfterRecallTests extends APITests {
+  private static Clock clock;
+
+  @BeforeClass
+  public static void setUpBeforeClass() throws Exception {
+    clock = Clock.fixed(Instant.now(), ZoneOffset.UTC);
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    // reset the clock before each test (just in case)
+    ClockManager.getClockManager().setClock(clock);
+  }
+
+  @Test
+  public void recallRequestWithNoPolicyValuesChangesDueDateToSystemDate()
+      throws InterruptedException,
+      ExecutionException,
+      TimeoutException,
+      MalformedURLException {
+    final IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    final IndividualResource requestServicePoint = servicePointsFixture.cd1();
+    final IndividualResource steve = usersFixture.steve();
+    final IndividualResource jessica = usersFixture.jessica();
+
+    final IndividualResource loan = loansFixture.checkOut(smallAngryPlanet, steve, DateTime.now(DateTimeZone.UTC));
+
+    final String originalDueDate = loan.getJson().getString("dueDate");
+
+    requestsFixture.placeHoldShelfRequest(smallAngryPlanet, jessica,
+        DateTime.now(DateTimeZone.UTC), requestServicePoint.getId(), "Recall");
+
+    final JsonObject storedLoan = loansStorageClient.getById(loan.getId()).getJson();
+
+    assertThat("due date is not the original date",
+        storedLoan.getString("dueDate"), not(originalDueDate));
+
+    final String expectedDueDate = new DateTime(ZonedDateTime.now(clock).toInstant().toEpochMilli(), DateTimeZone.UTC).toString(ISODateTimeFormat.dateTime());
+    assertThat("due date is the current date",
+        storedLoan.getString("dueDate"), is(expectedDueDate));
```

Comment:
I wonder, Is a message described what went wrong or just described a check. I think it should be 
 `assertThat("due date is not the current date", storedLoan.getString("dueDate"), is(expectedDueDate));`
because when it's failed this message should give you a clue. 
What do you think?

Reply:
Yeah, the JavaDoc sucks for `assertThat`. I always thought it was a message about what the assertion is trying to assert, but looking at the hamcrest code, it appears to be the reason failure. Whoops! Thanks for getting me to investigate. I'll update the messages.